### PR TITLE
Ability to mark words as learnt

### PR DESCRIPTION
**Pull request by rohankatyal29**
Words can be now marked as learnt
Words marked as learnt aren't translated
screen shot 2016-06-11 at 11 24 42 am

screen shot 2016-06-11 at 11 25 43 am

---- Minor Edits -----

- Changed note displaying hex values into rgba on the options page 

- Fixed bug causing empty values to be displayed on the options page
screen shot 2016-06-11 at 11 06 49 am

- Suggested changes in the class functions

### CODE REVIEW

File: lib/scripts/controllers/options.js [hunk 1]
```diff
@@ -56,8 +59,13 @@ export class OptionCtrl {
       console.log(data);
       this.patterns = JSON.parse(data.savedPatterns);
       this.key = data.yandexTranslatorApiKey;
+
       this.blacklistedWords = data.userBlacklistedWords.slice(1, -1).split('|');
+      this.blacklistedWords = this.blacklistedWords.filter(function(n){ return n != '' });
+
       this.blacklistedWebsites = data.blacklist.slice(1, -1).split('|');
+      this.blacklistedWebsites = this.blacklistedWebsites.filter(function(n){ return n != '' });
```
Comment (ankit-m) [inline lib/scripts/controllers/options.js]: @rohankatyal29 why are these required? Can't we make sure that we don't blacklist blank words. That would be a better check. What do you think?

Comment (Ceilican) [inline lib/scripts/controllers/options.js]: I agree with @ankit-m on this.

**Comment by Ceilican** [comment]
The user should be informed that learnt words won't be translated. I suggest that we add a small question mark icon next to "Learnt Words" which would display a tooltip that would explain what this feature does when the user clicks on it or hover over it. What do you think?

File: lib/scripts/controllers/options.js (hunk 1 above)
Comment (rohankatyal29) [inline lib/scripts/controllers/options.js]: The blank word was being added because of the slicing and splitting not
because of blacklisting empty words.

Without merging my code, delete all the blacklisted words and refresh the
options page you'll see the blank entry.

On Saturday 11 June 2016, Bruno <notifications@github.com> wrote:

> In lib/scripts/controllers/options.js
> <https://github.com/OiWorld/MindTheWord/pull/131#discussion_r66701759>:
>
> >        this.blacklistedWebsites = data.blacklist.slice(1, -1).split('|');
> > +      this.blacklistedWebsites = this.blacklistedWebsites.filter(function(n){ return n != '' });
>
> I agree with @ankit-m <https://github.com/ankit-m> on this.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OiWorld/MindTheWord/pull/131/files/2313eec4b773f9ed65a3413141e4e4a8b38c2611#r66701759>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACVryqUwnC7FygcAFTbWQiJ2MTDs99nbks5qKlPLgaJpZM4IzexP>
> .
>


-- 
Regards,
Rohan Katyal

Please ignore the typos and brevity, email sent using a small form factor
device

**Comment by rohankatyal29** [comment]
Makes sense, will add this!

On Saturday 11 June 2016, Bruno <notifications@github.com> wrote:

> The user should be informed that learnt words won't be translated. I
> suggest that we add a small question mark icon next to "Learnt Words" which
> would display a tooltip that would explain what this feature does when the
> user clicks on it or hover over it. What do you think?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OiWorld/MindTheWord/pull/131#issuecomment-225339954>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACVryh11bYpLbNGMx4oolu2PH7m-Z_U3ks5qKlRbgaJpZM4IzexP>
> .
>


-- 
Regards,
Rohan Katyal

Please ignore the typos and brevity, email sent using a small form factor
device

File: lib/scripts/controllers/options.js (hunk 1 above)
Comment (Ceilican) [inline lib/scripts/controllers/options.js]: Then this corner case should preferably be handled in line 66, instead of fixed in line 67...  Can't the blacklist be sliced and split in a more careful way, so that this problematic corner case doesn't occur?
